pkg/acceptance/barecluster/localcluster.go

```diff
@@ -425,15 +513,139 @@ func (n *Node) Start() {
 		n.Lock()
 		n.cmd = nil
 		n.Unlock()
-	}(n.cmd)
+	}(n.cmd, n.cmd.Process.Pid)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		n.waitUntilLive()
+	}()
+}
+
+func portFromURL(rawURL string) (string, *url.URL, error) {
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return "", nil, err
+	}
+
+	_, port, err := net.SplitHostPort(u.Host)
+	return port, u, err
+}
+
+func makeDB(url string, numWorkers int, dbName string) *gosql.DB {
+	conn, err := gosql.Open("postgres", url)
+	if err != nil {
+		log.Fatal(context.Background(), err)
+	}
+	if numWorkers == 0 {
+		numWorkers = 1
+	}
+	conn.SetMaxOpenConns(numWorkers)
+	conn.SetMaxIdleConns(numWorkers)
+	return conn
+}
+
+func (n *Node) waitUntilLive() {
+	ctx := context.Background()
+	opts := retry.Options{
+		InitialBackoff: time.Millisecond,
+		MaxBackoff:     500 * time.Millisecond,
+		Multiplier:     2,
+	}
+	for r := retry.Start(opts); r.Next(); {
+		urlBytes, err := ioutil.ReadFile(n.listeningURLFile())
+		if err != nil {
+			continue
+		}
+
+		var pgURL *url.URL
+		_, pgURL, err = portFromURL(string(urlBytes))
+		if err != nil {
+			log.Info(ctx, err)
+			continue
+		}
+
+		if n.cfg.RPCPort == 0 {
+			n.Lock()
+			n.rpcPort = pgURL.Port()
+			n.Unlock()
+		}
+
+		pgURL.Path = n.cfg.DB
+		n.Lock()
+		n.pgURL = pgURL.String()
+		n.Unlock()
+
+		pidBytes, err := ioutil.ReadFile(n.pidFile())
+		if err != nil {
+			log.Info(ctx, err)
+			continue
+		}
+
+		pid := strings.TrimSpace(string(pidBytes))
```

Comment:
Is this different from `n.cmd.Process.Pid`? If yes, it deserves a comment. Seems like they will be the same unless `--background` is specified which I don't think you're using, but perhaps I missed that.